client/dashboard/sites/overview/plan-card.tsx

```diff
@@ -0,0 +1,43 @@
+import { useQuery } from '@tanstack/react-query';
+import { cloneElement } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import { wordpress } from '@wordpress/icons';
+import { siteCurrentPlanQuery } from '../../app/queries/site-plans';
+import { sitePurchaseQuery } from '../../app/queries/site-purchases';
+import OverviewCard from '../overview-card';
+import type { Site, Plan, Purchase } from '../../data/types';
+
+export default function PlanCard( { site }: { site: Site } ) {
+	const { data: plan, isLoading: isLoadingPlan } = useQuery( siteCurrentPlanQuery( site.ID ) );
+	const { data: purchase, isLoading: isLoadingPurchase } = useQuery( {
+		...sitePurchaseQuery( site.ID, plan?.id ?? '' ),
+		enabled: !! plan?.id,
+	} );
+
+	const icon = cloneElement( wordpress, {
+		style: { fill: 'var( --wp-admin-brand-color )' },
+	} );
+
+	return (
+		<OverviewCard
+			title={ __( 'Plan' ) }
+			icon={ icon }
+			heading={ site.plan?.product_name_short || '\u00A0' }
+			description={ getCardDescription( plan, purchase ) }
+			trackId="plan"
+			variant={ isLoadingPlan || isLoadingPurchase ? 'loading' : undefined }
+		/>
+	);
+}
+
+function getCardDescription( plan?: Plan, purchase?: Purchase ) {
+	if ( plan?.product_slug === 'free_plan' ) {
+		return __( 'Upgrade to access all hosting features.' );
+	}
+
+	if ( purchase?.expiry_message ) {
+		return purchase.expiry_message;
```

Comment:
Note that this text can say "Expires on ..." even though the mockups say "Renews on ...". My understanding is that the specific verbiage we use has special logic. If you have a credit card saved then we can say "Renews on". But if you don't have auto-renew enabled, then the verb is going to be "Expires".